gobblin-service/src/test/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelperTest.java

```diff
@@ -66,4 +100,127 @@ public void testSkipAddingFlowExecutionIdWhenPresent() {
+            .withValue(ConfigurationKeys.FLOW_GROUP_KEY, ConfigValueFactory.fromAnyRef(flowGroup))
+            .withValue(ConfigurationKeys.FLOW_NAME_KEY, ConfigValueFactory.fromAnyRef(flowName))
+            .withValue(ConfigurationKeys.GOBBLIN_FLOW_FINISH_DEADLINE_TIME_UNIT, ConfigValueFactory.fromAnyRef(TimeUnit.MILLISECONDS.name()))
+            .withValue(ConfigurationKeys.GOBBLIN_FLOW_FINISH_DEADLINE_TIME, ConfigValueFactory.fromAnyRef(flowFinishDeadline))
+            .withValue(ConfigurationKeys.SPECEXECUTOR_INSTANCE_URI_KEY, ConfigValueFactory.fromAnyRef(
+                MySqlDagManagementStateStoreTest.TEST_SPEC_EXECUTOR_URI)));
+    dag.getNodes().forEach(node -> node.getValue().setFlowStartTime(flowStartTime));
+    this.dagManagementStateStore.addDag(dag);
+
+    Assert.assertFalse(FlowCompilationValidationHelper.isFlowRunning(flowGroup, flowName, this.dagManagementStateStore));
+  }
+
+  @Test
+  public void testConcurrentFlowPreviousFlowWithNonTerminalStatusRunningWithinFlowFinishDeadline()
+      throws IOException, URISyntaxException {
+    List<FlowStatus> list = new ArrayList<>();
+    String flowGroup = "fg";
+    String flowName = "fn";
+    long previousFlowExecutionId = 12345L;
+    long flowFinishDeadline = 10000L;
+    long flowStartTime = System.currentTimeMillis();  // giving test flowFinishDeadline to finish
+    JobStatus jobStatus = JobStatus.builder().flowGroup(flowGroup).flowName(flowName).flowExecutionId(previousFlowExecutionId)
+        .jobName(JobStatusRetriever.NA_KEY).jobGroup(JobStatusRetriever.NA_KEY).eventName(ExecutionStatus.RUNNING.name())
+        .build();
+    Iterator<JobStatus> jobStatusIterator = Lists.newArrayList(jobStatus).iterator();
+    list.add(new FlowStatus(flowName, flowGroup, previousFlowExecutionId, jobStatusIterator, ExecutionStatus.RUNNING));
+    when(this.dagManagementStateStore.getAllFlowStatusesForFlow(anyString(), anyString())).thenReturn(list);
+    Dag<JobExecutionPlan> dag = DagManagerTest.buildDag("1", previousFlowExecutionId,
+        DagProcessingEngine.FailureOption.FINISH_ALL_POSSIBLE.name(), 5, "user5", ConfigFactory.empty()
+            .withValue(ConfigurationKeys.FLOW_GROUP_KEY, ConfigValueFactory.fromAnyRef(flowGroup))
+            .withValue(ConfigurationKeys.FLOW_NAME_KEY, ConfigValueFactory.fromAnyRef(flowName))
+            .withValue(ConfigurationKeys.GOBBLIN_FLOW_FINISH_DEADLINE_TIME_UNIT, ConfigValueFactory.fromAnyRef(TimeUnit.MILLISECONDS.name()))
+            .withValue(ConfigurationKeys.GOBBLIN_FLOW_FINISH_DEADLINE_TIME, ConfigValueFactory.fromAnyRef(flowFinishDeadline))
+            .withValue(ConfigurationKeys.SPECEXECUTOR_INSTANCE_URI_KEY, ConfigValueFactory.fromAnyRef(
+                MySqlDagManagementStateStoreTest.TEST_SPEC_EXECUTOR_URI)));
+    dag.getNodes().forEach(node -> node.getValue().setFlowStartTime(flowStartTime));
+    this.dagManagementStateStore.addDag(dag);
```

Comment:
this DMSS mock config is so extensive that it really overshadows the intent of these tests.  how about abstracting into:
```
public static void insertFlowIntoDMSSMock(String flowGroup, String flowName, long flowExecId, ExecutionStatus currFlowAndJobStatus, Config additionalConfigs)
```
NOTES:
  * choose random username
  * set flowStartTime == flowExecId
  * use additionalConfigs as a fallback after adding: FLOW_GROUP_KEY, FLOW_NAME_KEY, and SPECEXECUTOR_INSTANCE_URI_KEY
  * always set GOBBLIN_FLOW_FINISH_DEADLINE_TIME_UNIT and GOBBLIN_JOB_START_DEADLINE_TIME_UNIT to TimeUnit.MILLISECONDS... even if it may not actually be necessary

with that, the setup and the assertion here are far simpler and easier to grasp:
```
insertFlowIntoDMSSMock(flowGroup, flowName, previousFlowExecutionId, ExecutionStatus.RUNNING,
  ConfigFactory.empty().
    withValue(ConfigurationKeys.GOBBLIN_FLOW_FINISH_DEADLINE_TIME, ConfigValueFactory.fromAnyRef(flowFinishDeadline)));

Assert.assertTrue(FlowCompilationValidationHelper.isFlowRunning(flowGroup, flowName, this.dagManagementStateStore));
```